rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1097,7 +1103,11 @@ stream {
             {{ else }}
             {{ $proxySetHeader }} X-Forwarded-For        $the_real_ip;
             {{ end }}
+            {{ if and $all.Cfg.UseForwardedHeaders (not $all.Cfg.UseForwardedHostHeader) }}
```

Comment:
Yeah, this is a little bit confusing. 

Existing behavior always sets `X-Forwarded-Host` and `Host` to `$best_http_host`. If we set `UseForwardedHeaders`, `$best_http_host` will be `X-Forwarded-Host` otherwise it will be the default value of the current backend.

Setting `UseForwardedHostHeader` prevents `X-Forwarded-Host` being set as the `Host` but we've still said we trust `X-Forwarded-*` by setting `UseForwardedHeaders`. As such the right behavior of the ingress seems to be to forward the `X-Forwarded-Host` value. I'm pretty sure this logic preserves this behavior:

>If we're trusting the X-Forwarded-* headers (`$all.Cfg.UseForwardedHeaders`) but **not** assigning `X-Forwarded-Host` to `Host` (`not $all.Cfg.UseForwardedHostHeader`) then we should forwarded the `X-Forwarded-Host` on. Otherwise we supply what was determined to be `$best_http_host`.

This makes `X-Forwarded-Host` be treated in the same way as `X-Forwarded-Proto` and `X-Forwarded-Port` that are just forwarded on to the backend.

Please let me know if this doesn't make sense or if I'm misunderstanding what the desired behavior should be!

Reply:
Perhaps changing the configuration option name would make this clearer? `OverrideHostHeader` perhaps?